web/react/stores/post_store.jsx

```diff
@@ -365,6 +331,24 @@ class PostStoreClass extends EventEmitter {
         this.postsInfo[id].atBottom = atBottom;
     }
 
+    deletePost(post) {
+        const postList = this.postsInfo[post.channel_id].postList;
+
+        if (isPostListNull(postList)) {
+            return;
+        }
+
+        if (post.id in postList.posts) {
+            // make sure to copy the post so that component state changes work properly
+            postList.posts[post.id] = Object.assign({}, post, {
+                message: '(message deleted)',
```

Comment:
Another thing @hmhealey do you think it's possible to remove the (message deleted) string from the PostStore? cause that has to be localized somehow, I was planning on setting that message when PostStore is loaded in the channel_view but can u think of a better way?